spec/std/http/server/server_spec.cr

```diff
@@ -504,6 +511,29 @@ module HTTP
         end
       end
     {% end %}
+
+    it "handles exception during SSL handshake (#6577)" do
+      server = LastExceptionHTTPServer.new { }
+      server_context, _ = ssl_context_pair
+      address = server.bind_tls "localhost", server_context
+
+      server_done = false
+      spawn do
+        server.listen
+        server_done = true
+      end
+
+      context = OpenSSL::SSL::Context::Client.new
+      socket = TCPSocket.new(address.address, address.port)
+      expect_raises(OpenSSL::SSL::Error) do
```

Comment:
Is there a specific reason why you removed the specific message matcher `certificate verify failed`?

Reply:
because the spec was failing in linux, but it turns out that the exception is not even raised. Need to double check in a vm...

Reply:
The spec I posted worked on my ubuntu 16.04 on WSL.

Perhaps try adding a few more `Fiber.yield` to make sure the error handling code is *really* executed.